src/decoder/lattice-faster-online-decoder.cc

```diff
@@ -135,24 +135,28 @@ typename LatticeFasterOnlineDecoderTpl<FST>::BestPathIterator LatticeFasterOnlin
   Token *tok = static_cast<Token*>(iter.tok);
   int32 cur_t = iter.frame, ret_t = cur_t;
   if (tok->backpointer != NULL) {
+    BaseFloat best_graph_cost = std::numeric_limits<BaseFloat>::infinity();
     ForwardLinkT *link;
     for (link = tok->backpointer->links;
          link != NULL; link = link->next) {
       if (link->next_tok == tok) { // this is the link to "tok"
-        oarc->ilabel = link->ilabel;
-        oarc->olabel = link->olabel;
-        BaseFloat graph_cost = link->graph_cost,
-            acoustic_cost = link->acoustic_cost;
-        if (link->ilabel != 0) {
-          KALDI_ASSERT(static_cast<size_t>(cur_t) < this->cost_offsets_.size());
-          acoustic_cost -= this->cost_offsets_[cur_t];
-          ret_t--;
```

Comment:
@mahsa7823 @danpovey 
might be a bug here: the for loop may hit "ret_t--" multiple times, but single call of traceback() shouldn't push back the timestamp more than once. (0 for non-emitting frame, 1 for emitting frame, in theory). The corrupted timestamp will trigger the above ASSERTION failure in some cases, which may further crash the decoder during runtime.

Reply:
Oops.  Can you please make a fix?

Reply:
No problem, I'll create a PR this week.